.github/workflows/tests.yml

```diff
@@ -0,0 +1,185 @@
+name: Tests
+
+on:
+  push:
+    branches: master
+  pull_request:
+    branches: master
+
+env:
+  CARGO_INCREMENTAL: 0
+  RUSTFLAGS: "-Dwarnings"
+
+jobs:
+  check-doc:
+    name: Doc deadlinks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          override: true
+      - run: cargo install cargo-deadlinks
+      - run: cargo deadlinks -- --features custom
+
+  main-tests:
+    name: Main tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        toolchain: [nightly, beta, stable, 1.34]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.toolchain }}
+          override: true
+      - run: cargo test
+      - run: cargo test --features std
+      - if: ${{ matrix.toolchain == 'nightly' }}
+        run: cargo build --benches
```

Comment:
Do you think we should have a job that actually runs the benchmarks? This could be done in a followup CL.

Reply:
I think it will be a waste of CI time.

Reply:
Seems reasonable